### PR TITLE
Use align_method in comp_method_FRAME

### LINKED ISSUE #20090: Mixed-dtype dataframe comparison with array raises incorrectly
This came up while going through some of statsmodels tests:

```
arr = np.random.randn(3, 2)
arr[:, 0] = [1, 2, 3]
df = pd.DataFrame(arr)
df[0] = df[0].astype(int)
>>> df == arr
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pandas/core/ops.py", line 1572, in f
    try_cast=False)
  File "pandas/core/frame.py", line 4021, in _combine_const
    try_cast=try_cast)
  File "pandas/core/internals.py", line 3644, in eval
    return self.apply('eval', **kwargs)
  File "pandas/core/internals.py", line 3538, in apply
    applied = getattr(b, f)(**kwargs)
  File "pandas/core/internals.py", line 1348, in eval
    t_shape=values.T.shape, oth_shape=other.shape))
ValueError: cannot broadcast shape [(3, 1)] with block values [(3, 2)]
```

I'd expect this to wrap the ndarray in a frame and return an all-True frame.

### PR DESCRIPTION
**Pull request by jbrockmendel**
Redux to #22880 @jorisvandenbossche 

Closes #20090 

Summary:
DataFrame comparison operations broadcasting is inconsistent with arithmetic operations broadcasting.  This PR fixes that.

### CODE REVIEW

**Comment by jreback** [comment]
i think was ok with this before, needs a rebase, @jorisvandenbossche

**Comment by jbrockmendel** [comment]
@jorisvandenbossche focusing here instead of PeriodArray for a bit.  Can you recap your concerns?

**Comment by jbrockmendel** [comment]
@jorisvandenbossche gentle ping.  You're the gate-keeper for this and #23113.

**Comment by jreback** [comment]
would also merge master here

**Comment by jreback** [comment]
lgtm. merging shorty.

File: doc/source/whatsnew/v0.24.0.txt [hunk 1]
```diff
@@ -567,6 +567,88 @@ Previous Behavior:
...
+operations (``==``, ``!=``, ...) was inconsistent with the behavior of
+arithmetic operations (``+``, ``-``, ...).  The behavior of the comparison
+operations has been changed to match the arithmetic operations in these cases.
+(:issue:`22880`)
+
+The affected cases are:
+
+- operating against a 2-dimensional ``np.ndarray`` with either 1 row or 1 column will now broadcast the same way a ``np.ndarray`` would (:issue:`23000`).
+- a list or tuple with length matching the number of rows in the :class:`DataFrame` will now raise ``ValueError`` instead of operating column-by-column (:issue:`22880`.
+- a list or tuple with length matching the number of columns in the :class:`DataFrame` will now operate row-by-row instead of raising ``ValueError`` (:issue:`22880`).
```
Comment (jreback) [inline doc/source/whatsnew/v0.24.0.txt]: does this case match numpy?

Comment (jreback) [reply doc/source/whatsnew/v0.24.0.txt]: ok then

**Comment by jreback** [comment]
thanks!

**Comment by jorisvandenbossche** [comment]
@jreback I know I have not been responsive here (I have been rather busy with thoroughly looking at all the EA related stuff), but I would appreciate if you didn't merge this without me looking at it (I was the original reason this got reverted). There are other ways to reach out to me.

**Comment by jreback** [comment]
this was blocking other things; you need to divide your time then

**Comment by jorisvandenbossche** [comment]
Then at least mention it is blocking other PRs, then I can try to prioritize it

**Comment by jreback** [comment]
you were pinged multiple times if u really want to block then put a review request

**Pull request merged**